File: lab9/aircraft/bomber.py
from aircraft.airplane import Airplane

class Bomber(Airplane):
    def __init__(self, model, capacity, bomb_capacity):
        """Ініціалізує бомбардувальник з моделлю, місткістю та місткістю бомб."""
        super().__init__(model, capacity)
        self.bomb_capacity = bomb_capacity

    def drop_bombs(self):
        """Метод, що імітує скидання бомб."""
        return f"{self.model} скидає {self.bomb_capacity}  бомб!"

    def info(self):
        """Повертає інформацію про бомбардувальник."""
        return f"{super().info()}, Місткість бомб: {self.bomb_capacity} "

File: lab3/lab3/src/CE306/Diadiuk/Lab3/Bomber.java
package CE306.Diadiuk.Lab3;

import java.io.IOException;

/**
 * Клас, що представляє бомбардувальник.
 */
public class Bomber extends Airplane implements BombDropper {
    private int bombCount;

    public Bomber(Engine engine, Wing leftWing, Wing rightWing, Cockpit cockpit, Tank tank, int bombCount) throws IOException {
        super(engine, leftWing, rightWing, cockpit, tank);
        this.bombCount = bombCount;
    }

    
    public void dropBombs() {
        if (bombCount > 0) {
            System.out.println("Бомба скинута!");
            bombCount--;
        } else {
            System.out.println("Усі бомби скинуті.");
        }
    }

    public void dropBomb() {
        dropBombs();
    }

    public int getBombCount() {
        return bombCount;
    }
}
